solana/src/main/java/com/solana/core/Transaction.kt

```diff
@@ -1,65 +1,450 @@
+// This is a code is a port from https://github.com/portto/solana-web3.kotlin
 package com.solana.core
 
-import com.solana.vendor.ShortvecEncoding
+import com.solana.vendor.*
 import org.bitcoinj.core.Base58
 import java.nio.ByteBuffer
-import java.util.*
+
+val DEFAULT_SIGNATURE = ByteArray(0)
+
+const val PACKET_DATA_SIZE = 1280 - 40 - 8
+
+const val SIGNATURE_LENGTH = 64
+
+class AccountMeta(
+    var publicKey: PublicKey,
+    var isSigner: Boolean,
+    var isWritable: Boolean
+) {
+    override fun toString(): String {
+        return "pubkey:${publicKey.toBase58()}, signer:$isSigner, writable:$isWritable"
+    }
+}
+
+class SerializeConfig(
+    val requireAllSignatures: Boolean = true,
+    val verifySignatures: Boolean = true
+)
+
+data class TransactionInstruction(
+    var programId: PublicKey,
+    var keys: List<AccountMeta>,
+    var data: ByteArray = ByteArray(0)
+)
+
+data class SignaturePubkeyPair(
+    var signature: ByteArray?,
+    val publicKey: PublicKey
+)
+
+class NonceInformation(
+    val nonce: String,
+    val nonceInstruction: TransactionInstruction
+)
 
 class Transaction {
-    private val message: Message = Message()
-    private val signatures: MutableList<String> = ArrayList()
+    var signatures = mutableListOf<SignaturePubkeyPair>()
+    val signature: ByteArray?
+        get() = signatures.firstOrNull()?.signature
+
     private lateinit var serializedMessage: ByteArray
-    fun addInstruction(instruction: TransactionInstruction): Transaction {
```

Comment:
Do we need to deprecate this method so consumers code wont break when they upgrade?